src/main/java/com/google/sps/servlets/EditEventServlet.java

```diff
@@ -0,0 +1,215 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import static com.google.sps.Utils.getParameter;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import com.google.sps.Firebase;
+import com.google.sps.Utils;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.logging.Logger;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/edit-event")
+public class EditEventServlet extends HttpServlet {
+
+  private static final Logger LOGGER = Logger.getLogger(LoadEventServlet.class.getName());
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String userToken = request.getParameter("userToken");
+    if (Firebase.isUserLoggedIn(userToken)) {
+      String userID = Firebase.authenticateUser(userToken);
+
+      Key keyRequested = getEventKey(request, "key");
+      Query query = new Query("Event", keyRequested);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Entity eventRequested = datastore.prepare(query).asSingleEntity();
+
+      Entity eventEntity = updateEvent(request, eventRequested);
+      datastore.put(eventEntity);
+    } else {
+      throw new IOException("Cannot edit an event while not logged in");
+    }
+
+    // Redirect back to the my-events HTML page.
+    response.sendRedirect("/my-events.html");
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+    Key keyRequested;
+    try {
+      keyRequested = getEventKey(request, "Event");
+      if (keyRequested != null) {
```

Comment:
Try to handle the corner case first to make the error handling more explicit and the code less nested.
```
if (keyRequested == null) {
  // error handling
}
// Normal processing
```

Reply:
Will do, thanks!